### DISCUSSION TITLE
BPM integrated waveguides - taper simulation

### DISCUSSION THREAD

**Discussion by brumthAZ** [ankrh/BPM-Matlab#3] [Q&A]
Hi! First of all, congratulations for your work, and thank you so much, it is just GREAT to have a tool like this at one's disposal.

Summarizing, I'm aiming at having a mode solver + BPM solution for photonic integrated waveguide, and I've been testing BPM-Matlab. Pretty straightforwardly, I've been able to define the typical rectangular cross-section for a (SiO2) buried integrated waveguide.

![iwg_crosssec](https://user-images.githubusercontent.com/117349521/199688884-b97dbc5a-17ce-4dc5-98fe-4ce88af9b150.png)

Then, I've tried to perform a simulation for a tapered section, where the horizontal dimension (x) of the waveguide grows adiabatically, while "y" keeps unchanged. However, the software is focused to fiber photonics where tapered sections are typically a radial homogeneus thing. For what I see, you have not implemented the simulation with this degree of freedom (i.e. discriminating dimension). 
Is there an easy modification to do it? I've been modifying some code here and there with no success :/

Many thanks in advance!

**Comment by ankrh** [comment]
Thanks for your interest in BPM-Matlab!

You are correct that the built-in tapering functionality will taper all dimensions, which is not suitable for your simulations.

You should be able to instead define a 3D refractive index, as demonstrated in Example14.m. This method is more flexible but comes at the cost of increased memory requirements and slower computation speed.

Try it out and let me know if it works.
